gflownet/envs/base.py

```diff
@@ -186,39 +189,75 @@ def get_parents(
         actions = []
         return parents, actions
 
-    def step(self, action: Tuple[int]) -> Tuple[List[int], Tuple[int], bool]:
+    def _pre_step(
+        self, action: Tuple[int], skip_mask_check: bool = False
+    ) -> Tuple[bool, List[int], Tuple[int]]:
         """
-        Executes step given an action.
+        Performs generic checks shared by the step() method of all environments.
 
         Args
         ----
         action : tuple
             Action from the action space.
 
+        skip_mask_check : bool
+            If True, skip computing forward mask of invalid actions to check if the
+            action is valid.
+
         Returns
         -------
+        do_step : bool
+            If True, step() should continue further, False otherwise.
```

Comment:
Should `do_step` be returned last? It's purely cosmetic but it would yield an interface similar to `step()` that returns `(state, action, valid)`

Reply:
I put it first because `do_step` is the main output of `_pre_step` and to actually make it distinct to `valid`. The value of `do_step` is very much related to `valid`, but semantically it is different. So I would prefer to leave it as currently is.